### PR TITLE
fix(pfd/fm): implement mda/dh/no dh logic

### LINKED ISSUE #8067: bug(pfd/fm): NODH Shows as NaN on FMA
### Aircraft Version

Development

### Build info

```json
b0dc480
```

### Describe the bug

image

### Expected behavior

Should show NODH.

### Steps to reproduce

Enter NODH on PERF APPR as decision height, and get to cruise phase and close enough to destination airport for it to appear on the FMA.

### References (optional)

_No response_

### Additional info (optional)

_No response_

### Discord Username (optional)

_No response_

### PR DESCRIPTION
**Pull request by tracernz**
Fixes #8067

## Summary of Changes

Implements proper MDA/DH/NO DH logic and fixes some related issues + refactoring to more conventional avionics framework style along the way. Also fixed a bug with selected heading which the linter found while I was in the file.

Notes: We have some awkward issues with the various Arinc429Register/Value/RegisterSubject types and getting data across the bus, piping it around etc. Needs a bit of thought and probably a lot of cleanup.

## References

Various.

## Additional context

Discord username (if different from GitHub):

## Testing instructions

Check that NO DH in the PERF APPR RADIO field correctly displays NO DH on the FMA. Check that BARO and RADIO mininima work okay on the FMA, altitude tape (for MDA, digital readout goes amber below MDA, and there's an amber marker on the alt tape), and RA (amber DH appears above the artificial horizon RA redout when below DH).
Check that the selected heading bug on the PFD heading tape works normally.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_PerformancePage.js [hunk 1]
```diff
@@ -806,7 +806,7 @@ class CDUPerformancePage {
         };
 
         let baroCell = "[\xa0\xa0\xa0]";
-        if (isFinite(mcdu.perfApprMDA)) {
+        if (mcdu.perfApprMDA !== null) {
```
Comment (Benjozork) [inline fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_PerformancePage.js:809]: **question:** What is the reasoning behind this change?

File: fbw-a32nx/src/systems/instruments/src/PFD/shared/PFDSimvarPublisher.tsx [hunk 2]
```diff
@@ -295,6 +302,14 @@ export enum PFDVars {
     trueTrackRaw = 'L:A32NX_ADIRS_IR_1_TRUE_TRACK',
     slatPosLeft = 'L:A32NX_LEFT_SLATS_ANGLE',
     fm1NavDiscrete = 'L:A32NX_FM1_NAV_DISCRETE',
+    fm1EisDiscrete2 = 'L:A32NX_FM1_EIS_DISCRETE_WORD_2',
```
Comment (Benjozork) [inline fbw-a32nx/src/systems/instruments/src/PFD/shared/PFDSimvarPublisher.tsx:305]: **question:** can't we put those in `FMBusPublisher`?

File: fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_PerformancePage.js (hunk 1 above)
Comment (tracernz) [reply fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_PerformancePage.js:809]: NaN is for the floating point unit, not for a placeholder value.

File: fbw-a32nx/src/systems/instruments/src/PFD/shared/PFDSimvarPublisher.tsx (hunk 2 above)
Comment (tracernz) [reply fbw-a32nx/src/systems/instruments/src/PFD/shared/PFDSimvarPublisher.tsx:305]: We don't appear to be using that in the PFD? We should do that as a separate refactor.

**Comment by auroraisluna** [comment]
QA Report

Discord: alepouna
Object of testing: #8069 
Tier of Testing : 1
Date : 25/6/2023

Testing Process:
- FMA BARO displays
- FMA RADIO displays
- MDA displays (amber marker, tape amber)
- DH shows
- PFD HDG tape also normal

Testing Results:
Passed

**Pull request merged**